core/src/main/scala/org/apache/spark/rpc/netty/Inbox.scala

```diff
@@ -200,6 +200,16 @@ private[netty] class Inbox(val endpointName: String, val endpoint: RpcEndpoint)
    * Calls action closure, and calls the endpoint's onError function in the case of exceptions.
    */
   private def safelyCall(endpoint: RpcEndpoint)(action: => Unit): Unit = {
+    def dealWithFatalException(fatal: Throwable): Unit = {
+      inbox.synchronized {
+        // Should reduce the number of active threads before throw exception
+        if (numActiveThreads != 0) {
```

Comment:
Why this check ?

Reply:
Yes logically this check is not necessary, I add it only for safety since we are going to minus the value.

Reply:
should be safer to use `numActiveThreads > 0`?

Reply:
@cloud-fan that would be better, thanks, fixed